src/plugins/settings-ui/components/SettingsModal.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  createSignal,
+  Switch,
+  Match,
+  For,
+  createEffect,
+  onCleanup,
+} from 'solid-js';
+
+import General from './settings/General';
+import Appearance from './settings/Appearance';
+import Plugins from './settings/Plugins';
+import Advanced from './settings/Advanced';
+import About from './settings/About';
+import { YtIcons } from '@/types/icons';
+import { t } from '@/i18n';
+
+interface SidebarItem {
+  icon: YtIcons;
+  id: string;
+}
+
+const sidebar: SidebarItem[] = [
+  { icon: 'yt-icons:settings', id: 'general' },
+  { icon: 'yt-icons:color_lens', id: 'appearance' },
+  { icon: 'yt-icons:emoji', id: 'plugins' },
+  { icon: 'yt-icons:service_toolbox', id: 'advanced' },
+  { icon: 'yt-icons:info', id: 'about' },
+];
+
+interface SettingsModalProps {
+  close: () => void;
+}
+
+export default ({ close }: SettingsModalProps) => {
```

Comment:
In Solid.js, you should not use object destructuring on props. If you destructure props, you will lose reactivity.

Reply:
Yeah I know, I just don't have reactive values in that context